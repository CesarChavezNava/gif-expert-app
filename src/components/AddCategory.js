import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const initialState = '';
    const [inputValue, setInputValue] = useState(initialState);

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories( categories => [...categories, inputValue]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputChange} />
            <h2>Add Category</h2>
        </form>
    )
}

AddCategory.prototype = {
    setCategories: PropTypes.func.isRequired
}
