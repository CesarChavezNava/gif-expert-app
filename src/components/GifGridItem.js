import React from 'react'

export const GifGridItem = ( { title, url} ) => {
    return (
        <div className="column is-4">
            <div className="card">
                <div className="card-image">
                    <figure className="image is-4by3">
                        <img src={url} alt={title} />
                    </figure>
                </div>
                <div className="card-content">
                    <div className="content">
                        <p>{title}</p>
                    </div>
                </div>
            </div>
        </div>
    )
}
