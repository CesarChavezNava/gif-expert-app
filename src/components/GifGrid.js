import React from 'react'
import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <div className="container is-fluid">
            { loading && <progress className="progress is-small is-primary" max="100">15%</progress> }
           <h3 className="subtitle">{category}</h3> 
           <div className="columns">
               {
                   images.map((image) => (
                    <GifGridItem key={image.id} {...image} />
                   ))
               }
           </div>
        </div>
    )
}

GifGrid.prototype = {
    category: PropTypes.string.isRequired
}