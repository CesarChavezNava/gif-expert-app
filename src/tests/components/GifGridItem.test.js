import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas del componente GifGridItem', () => {
    const title = 'crybaby GIF'
    const url = 'https://media3.giphy.com/media/ygAmzS6qTOkzxG0VsB/giphy.gif?cid=ef5ad807tpsqr3euuhgy4cxigprybtps0f2qg0y73q8x3cqq&rid=giphy.gif';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p'); 
        expect(p.text().trim()).toBe(title);
    });
    

    test('Debe de tener la imagen igual a la url', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe tener la clase de animacion animate__pulse', () => {
        const div = wrapper.find('.column');
        const classes = div.prop('className');

        expect(classes).toMatch('animate__pulse')
    })
    
});
