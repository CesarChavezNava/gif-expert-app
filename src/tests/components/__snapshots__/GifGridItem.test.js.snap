// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="column is-4 animate__animated animate__pulse"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <figure
        className="image is-4by3"
      >
        <img />
      </figure>
    </div>
    <div
      className="card-content"
    >
      <div
        className="content"
      >
        <p />
      </div>
    </div>
  </div>
</div>
`;
